control-plane/src/main.rs

```diff
@@ -160,36 +162,63 @@ async fn tcp_server() -> Result<()> {
         .ok()
         .and_then(|raw_data_plane_version| semver::Version::parse(&raw_data_plane_version).ok());
 
+    let sem = Arc::new(Semaphore::new(MAX_REQS_MINUTE));
+    let permit_creator = sem.clone();
+
+    tokio::spawn(async move {
+        loop {
+            let to_add = MAX_REQS_MINUTE.checked_sub(permit_creator.available_permits());
+
+            println!("{:?} requests made in last 60 seconds", to_add.unwrap_or(0));
+
+            if let Some(to_add) = to_add {
+                println!("Adding {to_add} permits");
+                permit_creator.add_permits(to_add);
+            }
+
+            sleep(Duration::from_secs(60)).await;
+        }
```

Comment:
We may be better off publishing metrics to the sidecar instead of logging here. Would be more powerful that querying logs